feature_engine/variable_manipulation.py

```diff
@@ -52,10 +111,26 @@ def _find_categorical_variables(X, variables=None):
         if len(X[variables].select_dtypes(exclude='O').columns) != 0:
             raise TypeError("Some of the variables are not categorical. Please cast them as object "
                             "before calling this transformer")
+
     return variables
 
 
-def _find_all_variables(X, variables=None):
+def _find_all_variables(X: pd.DataFrame, variables: List[str]=None) -> List[str]:
+    """
+    Takes Pandas DataFrame and extracts all variables if not provided corresponding list.
```

Comment:
I would re-phrase to "If None, captures all variables in the dataframe in a list. If user enters variable names list, it returns the list" or similar. The first sentence is not totally clear imho.